File: JS/twoSum.js
function isPair(arr1, twoSum) {
  for (let i = 0; i < arr1.length; i++) {
    let l = 0; //left pointer
    let r = arr1.length - 1; //right pointer
    while (l < r) {
      if (arr1[l] + arr1[r] === twoSum) {
        return true;
      } else if (arr1[l] + arr1[r] > twoSum) {
        r--;
      } else {
        l++;
      }
    }
  }
  return false;
}
let arr1 = [1, 2, 3, 4, 5, 6, 7];
let twoSum = 9; //true if two sum exists
console.log(isPair(arr1, twoSum));

File: java/TwoSum.java
import java.util.Arrays;
import java.util.Map;
import java.util.HashMap;
public class Main
{
	public static int[] twoSum(int[] nums, int target) {
		Map<Integer, Integer> numMap = new HashMap<>();
		for (int i = 0; i < nums.length; i++) {
			int complement = target - nums[i];
			if (numMap.containsKey(complement)) {
				return new int[] { numMap.get(complement), i };
			}
			numMap.put(nums[i], i);
		}
		throw new IllegalArgumentException("No two sum solution");
	}
	public static void main(String[] args) {
		int[] nums= {1,2,3,4,5,6};
		System.out.println(Arrays.toString(twoSum(nums,7)));
	}
}
